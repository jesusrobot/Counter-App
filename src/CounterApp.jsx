import { useState } from 'react'

function CounterApp({ init }) {
  const [ counter, setCounter ] = useState( init )

  return (
    <div className='counter'>
      <h1>Contador</h1>
      <h2 className='counter_display'>{ counter }</h2>
      <div className='counter_buttons'>
        <button onClick={ () => setCounter(counter + 1) }>Agregar</button>
        <button onClick={ () => setCounter(init) }>Reestablecer</button>
        <button onClick={ () => setCounter(counter - 1) }>Quitar</button>
      </div>
    </div>
  )
}

export default CounterApp
