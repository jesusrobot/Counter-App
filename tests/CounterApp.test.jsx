import { fireEvent, render, screen } from "@testing-library/react"
import CounterApp from "../src/CounterApp"

describe('Pruebas en el componente <CounterApp />', () => { 

  const INIT_VALUE = 0

  test('Debe de hacer match con un ultima snapshot', () => {
    const { container } = render(<CounterApp init={ INIT_VALUE } />)

    expect( container ).toMatchSnapshot()
  })

  test('Debe de mostrar el valor inicial pasado por props', () => {
    render(<CounterApp init={ INIT_VALUE }/>)
    
    expect(screen.getByText(INIT_VALUE)).toBeTruthy()
  })

  test('Debe incrementar el counter al hacer click en el btn "Agregar"', () => {
    render(<CounterApp init={ INIT_VALUE } />)

    fireEvent.click(screen.getByText('Agregar'))
    
    expect(screen.getByText(INIT_VALUE + 1)).toBeTruthy()
  })

  test('Debe decrementar el counter al hacer click en el btn "Quitar"', () => {
    render(<CounterApp init={ INIT_VALUE } />)

    fireEvent.click(screen.getByText('Quitar'))
    
    expect(screen.getByText(INIT_VALUE - 1)).toBeTruthy()
  })

  test('Debe reiniciar el counter al hacer click en el btn "Reestablecer"', () => {
    render(<CounterApp init={ INIT_VALUE } />)

    fireEvent.click(screen.getByText('Agregar'))
    fireEvent.click(screen.getByText('Agregar'))
    fireEvent.click(screen.getByText('Reestablecer'))
    
    expect(screen.getByText(INIT_VALUE)).toBeTruthy()
  })
})