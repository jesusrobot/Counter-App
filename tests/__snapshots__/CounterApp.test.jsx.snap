// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CounterApp /> Debe de hacer match con un ultima snapshot 1`] = `
<div>
  <div
    class="counter"
  >
    <h1>
      Contador
    </h1>
    <h2
      class="counter_display"
    >
      0
    </h2>
    <div
      class="counter_buttons"
    >
      <button>
        Agregar
      </button>
      <button>
        Reestablecer
      </button>
      <button>
        Quitar
      </button>
    </div>
  </div>
</div>
`;
